Tests/scripts/validate_files_structure.py

```diff
@@ -682,6 +703,203 @@ def str2bool(v):
+
+
+def get_all_diff_text_files(branch_name, is_circle):
+
+    if is_circle:
+        branch_changed_files_string = run_git_command("git diff --name-status origin/master...{}".format(branch_name))
+        text_files_list = get_diff_text_files(branch_changed_files_string)
+
+    else:
+        local_changed_files_string = run_git_command("git diff --name-status --no-merges HEAD")
+        text_files_list = get_diff_text_files(local_changed_files_string)
+
+    return text_files_list
+
+
+def search_potential_secrets(secrets_file_paths):
+    """
+
+    :param secrets_file_paths: paths of files that are being commited to git repo
+    :return: dictionary of strings sorted by file name for secrets found in files
+    """
+    # Get generic white list set
+    with io.open('./Tests/secrets_white_list.json', mode="r", encoding="utf-8") as secrets_white_list_file:
+        secrets_white_list = set(json.load(secrets_white_list_file))
+
+    secrets_found = {}
+
+    for file_path in secrets_file_paths:
+        file_name = file_path.split('/')[-1]
+        high_entropy_strings = []
+        regex_secrets = []
+
+        # if py file, search for yml in order to retrieve temp white list
+        file_path_temp, file_extension = os.path.splitext(file_path)
+        if file_extension == '.py':
+            yml_file_contents = retrieve_related_yml(file_path_temp)
+
+        # Open each file, read its contents in UTF-8 encoding to avoid unicode characters
+        with io.open('./' + file_path, mode="r", encoding="utf-8") as file:
+            file_contents = file.read()
+
+            # Add all context output paths keywords to whitelist temporary
+            temp_white_list = create_temp_white_list(yml_file_contents if yml_file_contents else file_contents)
+            secrets_white_list = secrets_white_list.union(temp_white_list)
+
+            # Search by lines after strings with high entropy as possibly suspicious
+            for line in set(file_contents.split('\n')):
+
+                # if detected disable-secrets comment, skip the line
+                if bool(re.findall(r'(disable-secrets-detection)', line)):
+                    file_contents = file_contents.replace(line, '')
+                    continue
+
+                # REGEX scanning for IOCs and false positive groups
+                potential_secrets, false_positives = regex_for_secrets(line)
+                for potential_secret in potential_secrets:
+                    if not any(white_list_string in potential_secret for white_list_string in secrets_white_list):
+                        regex_secrets.append(potential_secret)
+                # added false positives into white list array before testing the strings in line
+                secrets_white_list = secrets_white_list.union(false_positives)
+
+                # calculate entropy for each string in the file
+                for string_ in line.split():
+                    string_ = string_.strip("\"()[],'><:;\\")
```

Comment:
why did you choose to strip by those characters ?

Reply:
Those chars are the common chars I've identified in my work and tests as problematic in messing with the entropy score, after I removed all of these the entropy balanced significantly.